app/src/sharedTest/java/org/oppia/android/app/parser/HtmlParserTest.kt

```diff
@@ -852,8 +852,10 @@ class HtmlParserTest {
     }
   }
 
-  private fun createDisplayLocaleImpl(context: OppiaLocaleContext): DisplayLocaleImpl =
-    DisplayLocaleImpl(context, machineLocale, androidLocaleFactory, formatterFactory)
+  private fun createDisplayLocaleImpl(context: OppiaLocaleContext): DisplayLocaleImpl {
+    val formattingLocale = androidLocaleFactory.createOneOffAndroidLocale(context)
```

Comment:
Fir this and other tests, are we only creating the locale synchronously because it is technically another situation where locale cannot be created in a timely manner?

Reply:
Actually, I did it this way to make things easier code-wise. More "correct" would be blocking on the async method like done for some tests, but it's logically close enough that it should be fine to use a mix.